src/main/java/ch/njol/skript/effects/EffTeleport.java

```diff
@@ -101,18 +99,23 @@ protected TriggerItem walk(Event e) {
 		Location loc = location.getSingle(e);
 		if (loc == null)
 			return next;
+		boolean unknownWorld = !loc.isWorldLoaded();
 
 		Entity[] entityArray = entities.getArray(e); // We have to fetch this before possible async execution to avoid async local variable access.
 		if (entityArray.length == 0)
 			return next;
 
 		if (!delayed) {
 			if (e instanceof PlayerRespawnEvent && entityArray.length == 1 && entityArray[0].equals(((PlayerRespawnEvent) e).getPlayer())) {
+				if (unknownWorld)
+					return next;
```

Comment:
I don't like that, in all other places we infer the world *from where they are now* because that's a reasonable assumption, but with respawning it isn't and I don't know if it's okay to substitute a different assumption, since it could mean people get stuck in like a wall in the wrong world in an infinite death loop.